### ISSUE TITLE
Engine crashes, will not let you load the engine until CS file is deleted

### ISSUE THREAD

**Issue by KMDES** [godotengine/godot#16689]
**Godot version:**

3.0 

**OS/device including version:**
Windows 10

**Issue description:**
When adding TitleScreen.cs from the (scripts folder) to any node in Titlescreen.tscn (scenes/titlescreen folder) when you either add it or try running the game it will crash the engine and the engine will crash on load every time after that until you delete the file TitleScreen.cs

As long as you leave TitleScreen.cs blank it will work, but if you even add one variable (int test = 1; for example) it will crash the engine.

It was working fine until I ran the game one time and it started doing this. Removed all the added code, still kept doing this. Removed all code, still will keep crashing. Adding the script to different nodes and scenes crashes the engine also. If I rename the script, it won't crash until I run the game for the first time, then it will crash the engine the same as the original file until deleted.

No errors are displayed except CrashHandlerException: Program Crashed

I have no idea what might be causing this, but hopefully the submitted project file will help.

**Steps to reproduce:**

**Minimal reproduction project:**

[GhostsOfTheWorldAbandoned.zip](https://github.com/godotengine/godot/files/1724232/GhostsOfTheWorldAbandoned.zip)

**Comment by akien-mga** [comment]
CC @neikeq

**Comment by Duehok** [comment]
It might have something to do with issue #16578 
If you try to attach the TitleScreen.cs without modification to a node that is not a Node2D, it is the same issue.
But the TitleScreen root node is really a Node2D so it is not only this problem.

I cannot check right now in Godot, but TitleScreen.cs has the class "public class TitleScreen : Node2D", and in the tscn file the node is called "titlescreen", all lower case: might the case mismatch be a cause?

**Comment by Duehok** [comment]
Yes, I tried yesterday with your project, your problem has nothing to do with mine (I think).

I see the exact same thing: crash to desktop of the editor when loading the project if the file is attached to any node in it.
godot 3.0 stable, Mono 5.4.1.7, win 8.1 .

**Comment by akien-mga** [comment]
Is this still reproducible in recent Godot 3.1 builds?

**Comment by Duehok** [comment]
So I did the following with the minimal reproduction project:
* Godot 3.0.6
- start the project without any script attached => start OK
- attach the problematic script (titlescreen.cs) and start the project => Godot hangs
So at least I still have the original symptoms, as expected.

* Godot 3.1 beta 2
- start the project without any script attached => start OK
- attach the problematic script and start the project => start then crash of the project, editor keeps running, and there is an error message in the debugger window!

The error is a stack overflow with a huge pile of call to something inside SceneChangerFramework.cs and titleScreen.cs.

I can run the project with debugging from Visual Studio Code, but none of the few breakpoints I tried were hit (they work on other projects, thumbs up for the devs!).

 I am not going to spend more time on that, the behavior in 3.1 beta 2 is a lot better than in 3.0 .
